Lib/pathlib.py

```diff
@@ -405,6 +271,33 @@ def __reduce__(self):
         # when pickling related paths.
         return (self.__class__, tuple(self._parts))
 
+    @classmethod
+    def _split_root(cls, part):
+        sep = cls._flavour.sep
+        rel = cls._flavour.splitdrive(part)[1].lstrip(sep)
+        anchor = part.removesuffix(rel)
+        if anchor:
+            anchor = cls._flavour.normpath(anchor)
+        drv, root = cls._flavour.splitdrive(anchor)
+        if drv.startswith(sep):
+            # UNC paths always have a root.
```

Comment:
Note that device paths that begin with "\\\\.\\" or "\\\\?\\" do not have an implicit root, except for "\\\\?\\UNC\\server\\share".

Reply:
Yep understood. Pathlib wrongly assumes that all UNC paths have a root, but that behaviour has been present from the first versions of pathlib, so I'm not introducing a regression here I don't think.

I will fix this, but it will take some time.